packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -164,20 +189,27 @@ const PipelineWrapper: React.FC<IProps> = ({
       .then((images: any) => {
         runtimeImages.current = images;
         const nodesCopy = JSON.parse(JSON.stringify(nodes));
+        const pipelinePropertiesCopy = JSON.parse(
+          JSON.stringify(updatedPipelineProperties)
+        );
         for (const node of nodesCopy) {
           node.properties.uihints.parameter_info[1].data.items = Object.values(
             runtimeImages.current
           );
         }
+        pipelinePropertiesCopy.properties.uihints.parameter_info[4].data.items = Object.values(
+          runtimeImages.current
+        );
         setUpdatedNodes(nodesCopy);
+        setUpdatedPipelineProperties(pipelinePropertiesCopy);
         changeHandler();
       })
       .catch(error => RequestErrors.serverError(error));
 
     return (): void => {
       currentContext.model.contentChanged.disconnect(changeHandler);
     };
-  }, []);
+  }, [updatedPipelineProperties]);
```

Comment:
This causes an infinitely looping call of this hook (and thus the api) since this var is set within this hook.